### PR TITLE
Docs to onboard contributor

### LINKED ISSUE #437: Support local configuration file
Like in the [code-review bot](https://github.com/mozilla/code-review/blob/master/bot/code_review_bot/cli.py#L72), we could use a local configuration file instead of Taskcluster secret in order to ease the development setup for new contributors

### PR DESCRIPTION
**Pull request by La0**
- Adds description + setup infos for Bot
- Adds setup infos for Frontend
- Support local config file for backend & bot (fixes #437)
- Document local config file in backend
- Add matrix link on all READMEs

A big blocker for contributor would be the mandatory GCS account: #439

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -37,14 +37,28 @@ docker run -v /tmp/ccov-redis:/data -p 6379:6379 redis
...
+  APP_CHANNEL: dev
+  GOOGLE_CLOUD_STORAGE: null
+```
+
+Using your local configuration:
 
 ```shell
-./run.sh
+LOCAL_CONFIGURATION=/path/to/code-coverage-conf.yml ./run.sh
+
```
Comment (marco-c) [inline backend/README.md]:
```suggestion
```

File: backend/README.md [hunk 2]
```diff
@@ -37,14 +37,28 @@ docker run -v /tmp/ccov-redis:/data -p 6379:6379 redis
...
 ```
 
-You can specify any other secret as:
+You can specify a firefox-ci Taskcluster secret instead using:
```
Comment (marco-c) [inline backend/README.md]:
```suggestion
You can specify a firefox-ci Taskcluster secret (e.g. project-relman/code-coverage/dev) instead using:
```

File: bot/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+It's built using Python 3.8 and few dependencies.
+
+## Developer setup
+
+Requirements:
+
+- Python 3.8
+- [virtualenv](https://virtualenv.pypa.io/en/stable/)
+- [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/)
```
Comment (marco-c) [inline bot/README.md]: In theory these are not required-required, you could run the project without them

File: bot/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+---
+common:
+  APP_CHANNEL: dev
+bot:
+  BACKEND_HOST: 'http://localhost:8000'
+  EMAIL_ADDRESSES: []
+  PHABRICATOR_TOKEN: api-xxx
+  PHABRICATOR_ENABLED: false
+  PHABRICATOR_URL: 'https://phabricator-dev.allizom.org/api/'
+	GOOGLE_CLOUD_STORAGE: null
```
Comment (marco-c) [inline bot/README.md]:
```suggestion
  GOOGLE_CLOUD_STORAGE: null
```

File: bot/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+Run the bot (in cron mode):
+
+```console
+mkdir -p build/{cache,work} # or elsewhere on your system
+code-coverage-cron--cache-root=build/cache --working-dir=build/work --local-configuration=path/to/code-coverage.yml
```
Comment (marco-c) [inline bot/README.md]:
```suggestion
code-coverage-cron --cache-root=build/cache --working-dir=build/work --local-configuration=path/to/code-coverage.yml
```

File: bot/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+The repo mode (with `code-coverage-repo` is harder to use, as it requires a Google Cloud Storage & Phabricator account.
```
Comment (marco-c) [inline bot/README.md]:
```suggestion
The repo mode (with `code-coverage-repo`) is harder to use, as it requires a Google Cloud Storage and a Phabricator account.
```

File: bot/code_coverage_bot/cli.py [hunk 7]
```diff
@@ -38,6 +40,12 @@ def setup_cli(ask_repository=True, ask_revision=True):
         help="Taskcluster Secret path",
         default=os.environ.get("TASKCLUSTER_SECRET"),
     )
+    parser.add_argument(
+        "--local-configuration",
+        help="Path to a local YAML configuration file",
+        type=open,
+        default=os.environ.get("TASKCLUSTER_SECRET"),
```
Comment (marco-c) [inline bot/code_coverage_bot/cli.py]: Mistake?

File: bot/code_coverage_bot/secrets.py [hunk 8]
```diff
@@ -15,10 +15,10 @@ class Secrets(dict):
     PHABRICATOR_TOKEN = "PHABRICATOR_TOKEN"
     GOOGLE_CLOUD_STORAGE = "GOOGLE_CLOUD_STORAGE"
 
-    def load(self, taskcluster_secret):
+    def load(self, taskcluster_secret=None, local_secrets=None):
         taskcluster_config.load_secrets(
             taskcluster_secret,
-            prefixes=["common", "code-coverage-bot"],
+            prefixes=["common", "code-coverage-bot", "bot"],
```
Comment (marco-c) [inline bot/code_coverage_bot/secrets.py]: For simplicity, we could choose just one of `code-coverage-bot` and `bot`

File: frontend/README.md [hunk 9]
```diff
@@ -1,7 +1,7 @@
 # Mozilla Code Coverage frontend
 This is a simple JavaScript application displaying information from the Mozilla code coverage backend (hosted as https://coverage.moz.tools)
 
-You can use it right now from  [mozilla.github.io/code-coverage-reports](https://mozilla.github.io/code-coverage-reports/)
+You can use it right now from  [coverage.moz.tools](https://coverage.moz.tools/)
```
Comment (marco-c) [inline frontend/README.md]: Finally :smile:

File: bot/code_coverage_bot/cli.py (hunk 7 above)
Comment (La0) [reply bot/code_coverage_bot/cli.py]: Yeap :cry:

File: bot/README.md (hunk 3 above)
Comment (La0) [reply bot/README.md]: Added an optional tag + comment to reflect this

**Pull request merged**